packages/enzyme/src/ShallowWrapper.js

```diff
@@ -180,6 +185,55 @@ function isPureComponent(instance) {
   return instance && instance.isPureReactComponent;
 }
 
+function getChildContext(node, hierarchy, renderer) {
+  const { instance, type: Component } = node;
+  if (!Component.childContextTypes) {
+    return {};
+  }
+  const childContext = instance.getChildContext();
+  Object.keys(childContext).forEach((key) => {
+    if (!(key in Component.childContextTypes)) {
+      const componentName = displayNameOfNode(node);
+      throw new Error(`${componentName}.getChildContext(): key "${key}" is not defined in childContextTypes.`);
+    }
+  });
+  if (typeof renderer.checkPropTypes === 'function') {
+    renderer.checkPropTypes(Component.childContextTypes, childContext, 'child context', hierarchy);
+  }
+  return childContext;
+}
+
+function spyOnGetChildContextInitialRender(nodes, adapter) {
+  if (
+    !isCustomComponentElement(nodes, adapter)
+    || !nodes.type.prototype
+    || typeof nodes.type.prototype.getChildContext !== 'function'
+  ) {
+    return null;
+  }
+
+  return spyMethod(nodes.type.prototype, 'getChildContext');
+}
+
+function privateSetChildContext(adapter, wrapper, instance, renderedNode, getChildContextSpy) {
+  const renderer = wrapper[RENDERER];
+  // We only support parent-based context.
```

Comment:
what does this mean for react 0.13? is everything the same as before the PR?

Reply:
Correct!

Reply:
Let's file a new issue after this is merged, then, to track adding support for that.